app/controllers/application_controller.rb

```diff
@@ -141,9 +141,9 @@ def authenticate_token
         @current_user = User.find(@session.uuid)
         SSOService.extend_session!(@session, @current_user)
       end
+      :already_attempted
     end
-  ensure
-    @authenticate_token_status = :already_attempted
+    @return_value
```

Comment:
Now the meat of this method (`authenticate_with_http_token `) will only execute once, but will still return the same value if called multiple times.